precompile/bls12-381/src/bls/mod.rs

```diff
@@ -0,0 +1,184 @@
+// This file is part of Darwinia.
+//
+// Copyright (C) 2018-2023 Darwinia Network
+// SPDX-License-Identifier: GPL-3.0
+//
+// Darwinia is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Darwinia is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Darwinia. If not, see <https://www.gnu.org/licenses/>.
+//
+// Inspired from https://github.com/w3f/apk-proofs/blob/main/bw6/src/bls/mod.rs
+
+// core
+use core::{borrow::Borrow, ops::Neg};
+// crates.io
+use ark_bls12_381::{
+	g2::Config as G2Config, Bls12_381, G1Affine, G1Projective, G2Affine, G2Projective,
+};
+use ark_ec::{
+	hashing::{
+		curve_maps::wb::WBMap, map_to_curve_hasher::MapToCurveBasedHasher, HashToCurve,
+		HashToCurveError,
+	},
+	models::short_weierstrass::Projective,
+	pairing::Pairing,
+	AffineRepr, CurveGroup,
+};
+use ark_ff::{field_hashers::DefaultFieldHasher, Zero};
+use ark_serialize::*;
+use ark_std::vec::Vec;
```

Comment:
Sometimes this is `ark_std::vec::Vec`. And sometimes this is `sp_std::vec::Vec`. Can we only use one here?

Reply:
Let me see the difference between them.

Reply:
They are same, use `sp_set` only